lib/oauth-db.js

```diff
@@ -38,65 +58,117 @@ OAuthDB.prototype.generateAuthCode = function(clientId, userId, scopes, expiresA
 
     var authCode = random.toString('hex');
 
-    this.authCodes[authCode] = {
-      client_id: clientId,
-      user_id: userId,
-      scopes: scopes,
-      expires_at: expiresAt
-    };
-
-    callback(null, authCode);
+    var saltedCode = crypto.createHash('sha256')
+      .update(authCode + salt)
+      .digest('hex');
+
+    executeQuery(queries.create.authCode, [
+      saltedCode,
+      clientId,
+      userId,
+      JSON.stringify(scopes),
+      expiresAt
+    ], function(err) {
+      if ( err ) {
+        return callback(err);
+      }
+
+      process.nextTick(function() {
```

Comment:
You can remove these `process.nextTick` wrappers for those queries. I originally did it that way since the in-memory DB i created would've been synchronous otherwise. Since doing the postgres query will always be asynchronous, no need for this!